components/layout_2020/dom_traversal.rs

```diff
@@ -411,6 +411,9 @@ where
     fn as_canvas(self) -> Option<(CanvasInfo, PhysicalSize<f64>)> {
         let node = self.to_threadsafe();
         let canvas_data = node.canvas_data()?;
+        if canvas_data.width == 0 || canvas_data.height == 0 {
+            return None;
```

Comment:
This doesn’t look right: it’ll generate a non-replaced box (as for `<div></div>`), but we want to generate a replaced box, which might have different layout behavior (even if zero-size). Somewhere under `layout_2020/display_list/` would be the right place for this change.